frontends/web/src/routes/new-settings/components/singledropdown/singledropdown.tsx

```diff
@@ -0,0 +1,66 @@
+import { FC } from 'react';
+import Select, { components, DropdownIndicatorProps, SingleValue as SingleValueType, SingleValueProps, OptionProps } from 'react-select';
+import styles from './singledropdown.module.css';
+
+type TOption = {
+    label: string;
+    value: any;
+}
+
+type TSelectProps = {
+    options: TOption[];
+    handleChange: (param?: any) => void;
+    defaultValue: TOption;
+}
+
+const DropdownIndicator: FC<DropdownIndicatorProps<any>> = (props) => {
+  return (
+    <components.DropdownIndicator {...props}>
+      <div className={styles.dropdown} />
+    </components.DropdownIndicator>
+  );
+};
+
+
+const Option: FC<OptionProps<any>> = (props) => {
+  const { label } = props.data;
+  return (
+    <components.Option {...props}>
+      <div className={styles.valueContainer}>
+        <span className={styles.optionValue}>{label}</span>
+      </div>
+    </components.Option>
+  );
+};
+
+const SingleValue: FC<SingleValueProps<any>> = (props) => {
+  const { label } = props.data;
+  return (
+    <components.SingleValue {...props}>
+      <div className={styles.valueContainer}>
+        <span className={styles.singleValue}>{label}</span>
+      </div>
+    </components.SingleValue>
+  );
+};
+
+
+export const SingleDropdown = ({ options, handleChange, defaultValue }: TSelectProps) => {
+  return (
+    <Select
+      className={styles.select}
+      classNamePrefix="react-select"
+      isSearchable={true}
+      defaultValue={defaultValue}
+      components={{ IndicatorSeparator: () => null, DropdownIndicator, SingleValue, Option }}
+      onChange={(selected) => {
+        if (selected) {
+          const value = (selected as SingleValueType<any>)?.value || '';
+          handleChange(value);
+        }
+      }
+      }
+      options={options}
+    />
+  );
+};
```

Comment:
there is many `any` here, I haven't really looked into but could you check if it is possible to type it?

Reply:
I'll see if we could use generics.

Reply:
it would be nice if it is more or less straight forward to do. else we can also merge as is.